### PR TITLE
Fix for ON states for climate entities #278

### LINKED ISSUE #278: State ON for climate entities
I think there is a potential bug in how climate entities state is determined - or it might be a 'feature' of my climate integration (https://github.com/mariusz-ostoja-swierczynski/tech-controllers). 

From [HA docs](https://developers.home-assistant.io/docs/core/entity/climate/):

```
hvac_mode - The current operation (e.g. heat, cool, idle). Used to determine state.
```
while:

```
hvac_action - The current HVAC action (heating, cooling)
```

So climate entity state is not really current state.... And all my climate entities (with `hvac_mode === heat`) are shown as ON in HA Fusion and there is no difference at all when they are Idle or actually currently heating.

I think for climate entities the actual state for the button should be determined from `hvac_action`.

Perhaps the following:

| Name                 | Description                                                         |
| -------------------- | ------------------------------------------------------------------- |
| `HVACMode.HEAT`      | The device is set to heat to a target temperature.                  |
| `HVACMode.COOL`      | The device is set to cool to a target temperature.                  |
| `HVACMode.HEAT_COOL` | The device is set to heat/cool to a target temperature range.       |
| `HVACMode.AUTO`      | The device is set to a schedule, learned behavior, AI.              |
| `HVACMode.DRY`       | The device is set to dry/humidity mode.                             |
| `HVACMode.FAN_ONLY`  | The device only has the fan on. No heating or cooling taking place. |

should be removed from `onStates` store and some specific condition added to https://github.com/matt8707/ha-fusion/blob/main/src/lib/Main/Button.svelte to determine climate state.

I can take a stab at a PR if this is real issue (not limited to my old integration).

### PR DESCRIPTION
**Pull request by anarion80**
Fixes #278

I left all the `heat`, `dry`, `cool` in onStates for now as I'm not sure if they are not used for something else to determine state.

Now only `hvac_action` is taken for climate state and then it shows as I would expect it to:
obraz

### CODE REVIEW

**Comment by matt8707** [comment]
States are determined here
https://github.com/matt8707/ha-fusion/blob/main/src/lib/Components/StateLogic.svelte

**Comment by anarion80** [comment]
Yes, but isn't state to **display** determined there? And my issue is with lighting the button up as if the entity is ON.

Unless you suggest to override the **display** state (currently the mode) as well and show there _current_ state from hvac_action? That would make sense and I can do that.

**Comment by anarion80** [comment]
Assuming we override the display state as well - HA already has translations for all hvac_action states except `preheating`. What do you expect in such situation? Should I just add the whole bunch as to `fetch.py` like so:

```python
(  # CLIMATE
        f"{COMPONENTS}/climate/translations/",
        [
            ...
            ("climate_action_heating", ["entity_component", "_", "state_attributes", "hvac_action", "state", "heating"]),
            ("climate_action_preheating", ["entity_component", "_", "state_attributes", "hvac_astion", "state", "preheating"]),
            ....
```

**Comment by matt8707** [comment]
Hmm, but what is the state of you climate entity? I think you're right that eg `heating` isnt an onState, but there is still 'on' in that list. So 'on' state would light up the button, and then in [StateLogic.svelte](https://github.com/matt8707/ha-fusion/blob/main/src/lib/Components/StateLogic.svelte) you can display the correct state from hvac_action or am I confused?

**Comment by anarion80** [comment]
The climate entity state is always the mode, so in my case `heat`, so it always shows as ON (button lighted up). The current state is only in entity attribute `hvac_action` and it never gets into entity state. So I think we need to update both the Button and StateLogic as it is currently in the PR - button state and state display taken from `hvac_action`. What would remain is the translation.

obraz

**Comment by matt8707** [comment]
The 'heat' state for an entity represents an 'on' state, so you would need to turn it 'off' to prevent the button from lighting up. What is the state, as opposed to the action, when the climate displays 'idle'?

Skärmavbild 2024-01-28 kl  13 47 54

Can you post attributes yaml in different scenarios?

**Comment by anarion80** [comment]
That's the thing - it is still `heat` as that is the mode set:
obraz
and as this is only heating element the mode stays in `heat` for the whole season lets say, obscuring the actual, current state.

File: src/lib/Stores.ts [hunk 1]
```diff
@@ -45,6 +45,15 @@ export const onStates = readable([
 	'gas'
 ]);
```
Comment (matt8707) [inline src/lib/Stores.ts:47]: stole this from ha source code

```js
export const climateHvacActionToMode = readable<Record<string, string>>({
	cooling: 'cool',
	drying: 'dry',
	fan: 'fan_only',
	preheating: 'heat',
	heating: 'heat',
	idle: 'off',
	off: 'off'
});
```

File: src/lib/Main/Button.svelte [hunk 2]
```diff
@@ -71,7 +74,11 @@
 	// icon is image if extension, e.g. test.png
 	$: image = icon?.includes('.');
```
Comment (matt8707) [inline src/lib/Main/Button.svelte:76]:
```js
	$: stateOn = $onStates?.includes(
		attributes?.hvac_action ? $climateHvacActionToMode?.[attributes?.hvac_action] : entity?.state
	);
```

File: src/lib/Main/Button.svelte [hunk 3]
```diff
@@ -8,6 +8,7 @@
 		lang,
 		motion,
 		onStates,
+		climateOnStates,
```
Comment (matt8707) [inline src/lib/Main/Button.svelte:11]: climateHvacActionToMode

File: src/lib/Main/Button.svelte [hunk 4]
```diff
@@ -36,6 +37,8 @@
 	/** display loader if no state change has occurred within `$motion`ms */
 	let delayLoading: ReturnType<typeof setTimeout> | null;
 
+	let stateOn: boolean;
```
Comment (matt8707) [inline src/lib/Main/Button.svelte:40]: remove

**Comment by matt8707** [comment]
Does these changes work for you?

**Comment by anarion80** [comment]
Yes, thanks 👍🏻

**Pull request merged**